src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -213,15 +214,25 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
          */
         return
             !expr.getName().startsWith("_")
-            && (
-                (   expr.getParent() instanceof SQLBinaryOpExpr
-                    && ((SQLBinaryOpExpr) expr.getParent()).getOperator() == SQLBinaryOperator.Equality
-                )
-                || expr.getParent() instanceof SQLInListExpr
-                || expr.getParent() instanceof SQLInSubQueryExpr
-                || expr.getParent() instanceof SQLSelectOrderByItem
-                || expr.getParent() instanceof MySqlSelectGroupByExpr
-            );
+            && (isValidIdentifier(expr) || checkIfNestedIdentifier(expr));
+    }
+
+    private boolean checkIfNestedIdentifier(SQLIdentifierExpr expr) {
+        return
+            expr.getParent() instanceof SQLMethodInvokeExpr
+            && ((SQLMethodInvokeExpr) expr.getParent()).getMethodName().equals("nested")
+            && isValidIdentifier((SQLMethodInvokeExpr) expr.getParent());
+    }
+
+    private boolean isValidIdentifier(SQLExpr expr) {
+        return
+            ( expr.getParent() instanceof SQLBinaryOpExpr
```

Comment:
np: seems like assigning `expr.getParent()` to variable would make this method body much simpler.

Reply:
+1

Reply:
Done.

Reply:
I think now you changed the semantics of the method, at least how I understand it it checks that the sub/child expression is a valid identifier, because when you passed in X, you checked for X.parent in the method, but now you pass in X, and check for X. I would prefer to keep the older code, with the extra variable assignment, or now you need to change the method name to properly reflect the semantics.

Reply:
Reverted with extra variable assignment to keep semantics intact.